src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -573,72 +573,99 @@ query_dfs_mount(const char *path)
 	return idx;
 }
 
-static void
+/* Discover fuse mount points from /proc/self/mounts and env DAOS_MOUNT_POINT.
+ * Return 0 for success. Otherwise return Linux errno. This function interception
+ * is still effective as long as num_dfs is not zero even if discover_daos_mount()
+ * returns a non-zero value.
+ */
+static int
 discover_daos_mount(void)
 {
-	int   idx, len_fs_root;
+	int   idx, len_fs_root, rc;
 	char *fs_root   = NULL;
 	char *pool      = NULL;
 	char *container = NULL;
 
 	/* Find the list of dfuse from /proc/mounts */
-	discover_dfuse();
+	rc = discover_dfuse();
+	if (rc) {
+		D_DEBUG(DB_ANY, "discover_dfuse() failed: %d (%s)\n", rc, strerror(rc));
+		return rc;
+	}
 
 	/* Add the mount if env DAOS_MOUNT_POINT is set. */
 	fs_root = getenv("DAOS_MOUNT_POINT");
 	if (fs_root == NULL)
-		return;
+		goto out;
 
 	if (num_dfs >= MAX_DAOS_MT) {
-		D_WARN("dfs_list[] is full already. Need to incease MAX_DAOS_MT.\n");
-		return;
+		D_FATAL("dfs_list[] is full already. Need to incease MAX_DAOS_MT.\n");
+		abort();
 	}
 
 	if (access(fs_root, R_OK)) {
 		D_DEBUG(DB_ANY, "no read permission for %s: %d (%s)\n", fs_root, errno,
 			strerror(errno));
-		return;
+		rc = EACCES;
+		goto out;
```

Comment:
@ashleypittman Do we prefer D_GOTO?
```suggestion
		D_GOTO(out, rc = EACCES);
```

Reply:
Fixed as you suggested. Thank you!